python/mxnet/gluon/block.py

```diff
@@ -191,6 +191,8 @@ def __setattr__(self, name, value):
                 for i, c in enumerate(self._children):
                     if c is existing:
                         self._children[i] = value
+            if isinstance(value, Block):
```

Comment:
This should be elif. Otherwise value will be registered twice.

Reply:
Also there should be a break after 193

Reply:
There's currently no guarantee that multiple children can't be the same instance, so not breaking is intentional.

Reply:
Cleaner way would be to use OrderedDict for children.